TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -905,6 +909,147 @@ private void WritePeptideResults()
+            foreach (var entry in variantPeptides)
+            {
+                var pwsm = entry.BestMatchingPeptides;
+                var nonvariantProtein = false;
+                foreach (var pep in pwsm)
+                {
+                    if (pep.Peptide.Protein.NonVariantProtein.Name == pep.Peptide.Protein.Accession)
+                    {
+                        nonvariantProtein = true;
+                    }
+                }
+                if (nonvariantProtein != true)
+                {
+                    confidentVariantPeps.Add(entry);
+                }
+            }
+
+            int savCount = 0;
+            int insertionCount = 0;
+            int deletionCount = 0;
+            int frameshiftCount = 0;
+            int stopGainCount = 0;
+            int stopLossCount = 0;
+            int exonLossCount = 0; ;
+            confidentVariantPeps.RemoveAll(p => p.FdrInfo.QValue > 0.01 || p.FdrInfo.QValueNotch > 0.01 || p.IsDecoy);            
+            List<PeptideSpectralMatch> modifiedVariantPeptides = confidentVariantPeps.Where(p => p.ModsIdentified.Count() > 0).ToList();
+            List<PeptideSpectralMatch> modifiedVariantSitePeptides = new List<PeptideSpectralMatch>();
+            foreach (var entry in modifiedVariantPeptides)
+            {
+                var variants = entry.BestMatchingPeptides.SelectMany(p => p.Peptide.Protein.AppliedSequenceVariations.Where(v => PsmTsvWriter.IntersectsWithVariation(p.Peptide, v, true) ==true).ToList()).ToList();
+                var mods = entry.ModsIdentified;
+                bool modifiedVariant = false;
+                foreach (var mod in mods)
+                {
+                    int residue = mod.Value;                    
+                    var modOnVariant = variants.Where(p => p.OneBasedBeginPosition >= residue && p.OneBasedEndPosition <= residue);
+                    if (modOnVariant.Count() > 0)
+                    {
+                        modifiedVariant = true;
+                    }
+                }
+                if (modifiedVariant == true)
+                {
+                    modifiedVariantSitePeptides.Add(entry);
+                }
+            }
+            foreach (var peptide in confidentVariantPeps)
+            {
+                var variantPWSM = peptide.BestMatchingPeptides.FirstOrDefault();
+                var variants = variantPWSM.Peptide.Protein.AppliedSequenceVariations;
+                var culture = CultureInfo.CurrentCulture;
+                foreach (var variant in variants)
+                {
+                    bool intersects = PsmTsvWriter.IntersectsWithVariation(variantPWSM.Peptide, variant, true);
+                    if(intersects == true)
+                    {
+                        if (culture.CompareInfo.IndexOf(variant.Description.Description, "missense_variant", CompareOptions.IgnoreCase)>=0)
```

Comment:
can one peptide have multiple? if not maybe switch/case is better?

Reply:
I am not sure if switch cases can be used when checking for sub-strings, but I will look into it.

Reply:
I do not believe that searching for a substring works well with switch cases